### PR TITLE
9: Remove emojis by default

### LINKED ISSUE #9: Plugin: Disable Emojis
@roborourke commented on [Tue Aug 14 2018](https://github.com/humanmade/platform-dev/issues/248)

It's a basic plugin that could be on by default, used on a bunch of internal projects. Basically removes default WP emoji fallbacks to images.

https://wordpress.org/plugins/disable-emojis/

---

@roborourke commented on [Wed Mar 20 2019](https://github.com/humanmade/platform-dev/issues/248#issuecomment-474889985)

Wouldn't use the plugin but we can add this to the CMS module. Moving there.

### PR DESCRIPTION
**Pull request by bradp**
Closes #9

### CODE REVIEW

File: inc/namespace.php [hunk 1]
```diff
@@ -51,6 +51,38 @@ function bootstrap() {
...
+	if ( $config['remove-emojis'] ) {
+		remove_action( 'wp_head', 'print_emoji_detection_script', 7 );
+		remove_action( 'admin_print_scripts', 'print_emoji_detection_script' );
+		remove_action( 'wp_print_styles', 'print_emoji_styles' );
+		remove_action( 'admin_print_styles', 'print_emoji_styles' );
+		remove_filter( 'the_content_feed', 'wp_staticize_emoji' );
+		remove_filter( 'comment_text_rss', 'wp_staticize_emoji' );
+		remove_filter( 'wp_mail', 'wp_staticize_emoji_for_email' );
+		add_filter( 'tiny_mce_plugins', 'disable_emojis_tinymce' );
+		add_filter( 'wp_resource_hints', __NAMESPACE__ . '\\disable_emojis_remove_dns_prefetch', 10, 2 );
```
Comment (rmccue) [inline inc/namespace.php]: Can we move these actions/etc into a `remove_emoji` function instead? This function is getting a bit long now.

File: inc/namespace.php [hunk 2]
```diff
@@ -51,6 +51,38 @@ function bootstrap() {
...
+	}
+}
+
+/**
+ * Remove emoji DNS prefetching.
+ *
+ * @param array  $urls          URLs for resources.
+ * @param string $relation_type Relation type.
+ * @return array                New array with resources.
```
Comment (rmccue) [inline inc/namespace.php]: Param docs shouldn't be aligned

File: inc/namespace.php [hunk 3]
```diff
@@ -51,6 +51,38 @@ function bootstrap() {
...
+ */
+function disable_emojis_remove_dns_prefetch( $urls, $relation_type ) {
```
Comment (rmccue) [inline inc/namespace.php]: Typehints for the params and return values please!

File: inc/namespace.php [hunk 4]
```diff
@@ -51,6 +51,38 @@ function bootstrap() {
...
+	if ( 'dns-prefetch' == $relation_type ) {
```
Comment (rmccue) [inline inc/namespace.php]: This can flipped to an early return instead.

File: inc/namespace.php [hunk 5]
```diff
@@ -51,6 +51,38 @@ function bootstrap() {
...
+		// Strip out any URLs referencing the WordPress.org emoji location
+		$emoji_svg_url_bit = 'https://s.w.org/images/core/emoji/';
+		foreach ( $urls as $key => $url ) {
+			if ( strpos( $url, $emoji_svg_url_bit ) !== false ) {
+				unset( $urls[$key] );
```
Comment (rmccue) [inline inc/namespace.php]: Lint error.

File: load.php [hunk 6]
```diff
@@ -23,6 +23,7 @@
 		'login-logo'    => '/vendor/altis/cms/assets/logo.svg',
 		'shared-blocks' => true,
 		'default-theme' => 'base',
+		'remove-emojis' => true,
```
Comment (rmccue) [inline load.php]: Can we rename to `remove-emoji`? 😬

Comment (roborourke) [reply load.php]: classic Ryan :)

File: inc/namespace.php (hunk 4 above)
Comment (roborourke) [reply inc/namespace.php]: Should this be a strict comparison?

**Pull request merged**